File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/view/preferences/LocationPreference.java
package com.byagowi.persiancalendar.view.preferences;

import android.content.Context;
import android.util.AttributeSet;

import androidx.preference.DialogPreference;

/**
 * persian_calendar
 * Author: hamidsafdari22@gmail.com
 * Date: 1/17/16
 */
public class LocationPreference extends DialogPreference {

    public LocationPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public void setSelected(String selected) {
        final boolean wasBlocking = shouldDisableDependents();
        persistString(selected);
        final boolean isBlocking = shouldDisableDependents();
        if (isBlocking != wasBlocking) notifyDependencyChange(isBlocking);
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/view/preferences/LocationPreference.kt
package com.byagowi.persiancalendar.view.preferences

import android.content.Context
import android.content.Intent
import android.util.AttributeSet

import com.byagowi.persiancalendar.Constants

import androidx.localbroadcastmanager.content.LocalBroadcastManager
import androidx.preference.DialogPreference

/**
 * persian_calendar
 * Author: hamidsafdari22@gmail.com
 * Date: 1/17/16
 */
class LocationPreference(context: Context, attrs: AttributeSet) : DialogPreference(context, attrs) {

  fun setSelected(selected: String) {
    val wasBlocking = shouldDisableDependents()
    persistString(selected)
    val isBlocking = shouldDisableDependents()
    if (isBlocking != wasBlocking) notifyDependencyChange(isBlocking)
    LocalBroadcastManager.getInstance(context)
        .sendBroadcast(Intent(Constants.LOCAL_INTENT_UPDATE_PREFERENCE))
  }
}
